src/differential_value_iteration/experiments/evaluation_convergence.py

```diff
@@ -0,0 +1,64 @@
+"""Runs a sweep over evaluation algorithms and prints results."""
+import functools
+from typing import Sequence
+
+import numpy as np
+from absl import app
+from absl import flags
+from differential_value_iteration.algorithms import rvi
+from differential_value_iteration.environments import micro
+from differential_value_iteration.environments import structure
+
+FLAGS = flags.FLAGS
+flags.DEFINE_integer('max_iters', 50000, 'Maximum iterations per algorithm.')
+flags.DEFINE_float('minimum_step_size', .001, 'Minimum step size.')
+flags.DEFINE_float('maximum_step_size', 1., 'Maximum step size.')
+flags.DEFINE_integer('num_step_sizes', 10, 'Number of step sizes to try.')
+flags.DEFINE_float('convergence_tolerance', 1e-7, 'Tolerance for convergence.')
+
+
+def run(environments: Sequence[structure.MarkovRewardProcess],
+    step_sizes: Sequence[float], max_iters: int, convergence_tolerance: float):
+  algorithm_constructors = []
+
+  # Create a constructor that only depends on params common to all algorithms.
+  rvi_algorithm = functools.partial(rvi.Evaluation, reference_index=0)
+  algorithm_constructors.append(rvi_algorithm)
+
+  for environment in environments:
+    initial_values = np.zeros(environment.num_states)
+    for algorithm_constructor in algorithm_constructors:
+      print(f'Running {algorithm_constructor.func} on {environment.name}')
+      for step_size in step_sizes:
+        converged = False
+        algorithm = algorithm_constructor(mrp=environment,
+                                          initial_values=initial_values,
+                                          step_size=step_size,
+                                          synchronized=True)
+        for i in range(max_iters):
+          changes = algorithm.update()
+          if np.sum(np.abs(changes)) <= convergence_tolerance and i > 1:
```

Comment:
@yiwan-rl @abhisheknaik96 
This explicitly does not factor in r_bar changes, from DVI (for example).
However, in DVI, r_bar changes seem to be directly proportionate to the value changes, so we should not be losing information to just look at the value changes.

Does this make sense to you folks? I have not looked at the other algs yet, so maybe r_bar will be important to check for convergence elsewhere.

Reply:
You're right that in case of DVI this information suffices. But perhaps in some algo we might (a) not only need to track the changes to r_bar, (b) but also check for the convergence of r_bar and values separately. So perhaps we'll need to return changes to both? What do you think @yiwan-rl ?

Reply:
I think you might be right. My plan is to iteratively add new algorithms to the evaluation and adapt/refactor the code as necessary.

I think I have converted all of the evaluation algorithms without the need to check r_bar explicitly - but perhaps there is a mistake or it will be necessary for the control experiments.

On the other end of the spectrum, we do now have a diverged method, which explicitly ensures that r_bar is not NaN or Inf.